### PR TITLE
Fix `ostree admin kargs edit-in-place` fails issue

### PR DESCRIPTION
**Pull request by HuijingHei**
Add func to set kernel arguments in place, instead of create new
deployment

Fix https://github.com/ostreedev/ostree/issues/2664

### CODE REVIEW

**Comment by HuijingHei** [comment]
Test locally, this works
```
[core@cosa-devsh ~]$ sudo ostree admin kargs edit-in-place --append-if-missing=test1 -v
OT: remounted /sysroot writable
OT: remounted /boot writable
OT: Deployment 3ceadf5fc86a898c654e324f3e66df34ae44de0648ef2a0a195e86751590d1a9.0 unlocked=0
[core@cosa-devsh ~]$ cat /boot/loader.1/entries/ostree*
title Fedora CoreOS 36.20220711.dev.0 (ostree:0)
version 1
options mitigations=auto,nosmt ignition.platform.id=qemu console=tty0 console=ttyS0,115200n8 $ignition_firstboot ostree=/ostree/boot.1/fedora-coreos/0de25aa2ccec0f65ae8c7735d8f195c29c14428828ec19c45802648956ffbe4a/0 root=UUID=17fc46f7-8529-4ce2-a9f9-bd0408a4b637 rw rootflags=prjquota boot=UUID=3750607a-99e7-4957-bcdb-9717120cab8d test1
linux /ostree/fedora-coreos-0de25aa2ccec0f65ae8c7735d8f195c29c14428828ec19c45802648956ffbe4a/vmlinuz-5.18.10-200.fc36.x86_64
initrd /ostree/fedora-coreos-0de25aa2ccec0f65ae8c7735d8f195c29c14428828ec19c45802648956ffbe4a/initramfs-5.18.10-200.fc36.x86_64.img
```

**Comment by HuijingHei** [comment]
Thanks for the help of @cgwalters , will try to add a test case to `tests/kolainst/destructive`

**Comment by HuijingHei** [comment]
Build ostree with this patch, and copy the installed dir to cosa override/rootfs, then build qcow2 image, run test

```
[coreos-assembler]$ kola run ext.ostree.destructive.kargs-edit-in-place.sh
=== RUN   ext.ostree.destructive.kargs-edit-in-place.sh
--- PASS: ext.ostree.destructive.kargs-edit-in-place.sh (54.89s)
PASS, output in tmp/kola/qemu-unpriv-2022-07-12-1257-54
```

**Review by cgwalters — APPROVED** [review]
Nice work!

**Comment by cgwalters** [comment]
Hmm though actually looks like this broke the unit test case:

```
error: opendir(loader.1/entries): No such file or directory
++ report_err
++ local exit_status=1
Unexpected nonzero exit status 1 while running: ${CMD_PREFIX} ostree admin kargs edit-in-place --append-if-missing=TESTARG=TESTVALUE --append-if-missing=ARGWITHOUTKEY testos:testos/buildmain/x86_64-runtime
```

File: src/libostree/ostree-sysroot-deploy.c [hunk 1]
```diff
@@ -3589,6 +3589,49 @@ ostree_sysroot_deployment_set_kargs (OstreeSysroot     *self,
...
+
+  OstreeBootconfigParser *new_bootconfig = ostree_deployment_get_bootconfig (deployment);
+  ostree_bootconfig_parser_set (new_bootconfig, "options", kargs_str);
+
+  g_autofree char *bootconf_name =
+    g_strdup_printf ("ostree-%d-%s.conf",
+                     self->deployments->len - ostree_deployment_get_index (deployment),
+                     ostree_deployment_get_osname (deployment));
+
+  g_autofree char *bootconfdir = g_strdup_printf ("loader.%d/entries", self->bootversion);
```
Comment (cgwalters) [inline src/libostree/ostree-sysroot-deploy.c:3622]: The bug has something to do with `self->bootversion` here I think in the unit test case.  But...it should be correct as is AFAICS...

Comment (HuijingHei) [reply src/libostree/ostree-sysroot-deploy.c:3622]: Add assert before https://github.com/ostreedev/ostree/blob/main/tests/test-admin-kargs.sh#L32, check `sysroot/boot/loader.1/entries` exists, can not find `loader.1/entries`, is this expected?
```
+ assert_has_dir sysroot/boot/loader.1/entries
+ test -d sysroot/boot/loader.1/entries

+ assert_has_dir loader.1/entries
+ test -d loader.1/entries
+ fatal 'Couldn'\''t find '\''loader.1/entries'\'''
```

Comment (cgwalters) [reply src/libostree/ostree-sysroot-deploy.c:3622]: OK running this under `strace`, what I see is:

```
openat(AT_FDCWD, "loader.1/entries", O_RDONLY|O_NOCTTY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
```

which implies that `self->boot_fd` is `-1`.  Looking...

**Comment by cgwalters** [comment]
https://github.com/ostreedev/ostree/pull/2672 is the more general fix for that bug.

**Comment by cgwalters** [comment]
(Also let's do the equivalent of https://github.com/ostreedev/ostree/pull/2673 here too)

**Comment by HuijingHei** [comment]
Thanks @cgwalters for the fix!

**Pull request merged**